### PR TITLE
deprecate hydrogen bond analysis in water bridge analysis to master

### PR DESCRIPTION
**Pull request by xiki-tempula**
deprecate hydrogen bond analysis in water bridge analysis.
The hydrogen bond analysis is not used in the original water bridge analysis and I have cleared some reference to the hydrogen bond analysis in the doc sections.
Move water bridge analysis from hbonds to hydorgenbonds

Related to #2913, #2739 and #2746

PR Checklist
------------
 - [x] Tests?
 - [x] Docs?
 - [x] CHANGELOG updated?
 - [x] Issue raised/referenced?

### CODE REVIEW

**Comment by IAlibay** [comment]
Couple of things maintenance things:

1) I've made this a target for 1.0.2, the code freeze is currently on for 1.0.1, so please don't merge until we release.
2) I've restarted the failed Travis job, hopefully that'll complete this time.

Quick question @xiki-tempula, did you make any changes to the moved waterbridge analysis code (or it is just a standard move)? Github has done the really annoying thing of doing +1.5k -1.5k so it's pseudo-impossible to review anything from the diff.

**Comment by xiki-tempula** [comment]
@IAlibay Nothing has changed in waterbridge analysis code. Just PEP8 changes.

**Review by orbeckst — APPROVED** [review]
Looks good (for 1.0.2). Thank you, @xiki-tempula !

**Comment by orbeckst** [comment]
@IAlibay you will merge this PR when 1.0.1 is out, right?

**Comment by IAlibay** [comment]
> @IAlibay you will merge this PR when 1.0.1 is out, right?

Yup, I've placed it as a 1.0.2 milestone so we don't forget about it.

**Comment by IAlibay** [comment]
@xiki-tempula if you can update against the current `master` (probably will need to move the changelog entry to 1.0.2), I think we're at a point where we can merge this.

**Comment by xiki-tempula** [comment]
@IAlibay So I have merged the master and updated the changelog.
It seems that the test failed for py27 as py27 importlib doesn't have reload.
I wonder if the master branch still needs to support py27?

**Comment by IAlibay** [comment]
> @IAlibay So I have merged the master and updated the changelog.
> It seems that the test failed for py27 importlib.reload is introduced from py33.
> I wonder if the master branch still needs to support py27?

Yes unfortunately it does. I think you should be able to use reload_module from six.moves instead?

**Comment by xiki-tempula** [comment]
@IAlibay I think everything is fixed now.

File: package/MDAnalysis/analysis/hydrogenbonds/wbridge_analysis.py
Comment (IAlibay) [inline package/MDAnalysis/analysis/hydrogenbonds/wbridge_analysis.py:674]: to ensure proper backwards compatibility we'll need sex.move's range and zip too (worth having a look for other deprecated py2 things, but I _think_ that's it?).

Probably worth adding absolute_import here too.

File: package/CHANGELOG
Comment (IAlibay) [inline package/CHANGELOG:29]: Do add yourself to the 1.0.2 author list.

**Review by IAlibay — CHANGES_REQUESTED** [review]
Just some thoughts on py2 backward compatibility, and then one changelog addition.

File: package/MDAnalysis/analysis/hydrogenbonds/wbridge_analysis.py
Comment (xiki-tempula) [reply package/MDAnalysis/analysis/hydrogenbonds/wbridge_analysis.py]: I have added them but given that tests have all passed so I think should be fine.

Comment (IAlibay) [reply package/MDAnalysis/analysis/hydrogenbonds/wbridge_analysis.py]: Cheers. Yeah, it's really just for the sake of "correctness".

**Review by IAlibay — APPROVED** [review]
Thanks @xiki-tempula I'll merge once CI returns green.

**Pull request merged**
